src/util/deprecation.h

```diff
@@ -0,0 +1,26 @@
+/*******************************************************************\
+
+Module:
+
+Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+#ifndef CPROVER_UTIL_DEPRECATION_H
```

Comment:
I think this file would be better named `deprecate.h`, as most other things in the folder are named as verbs and not nouns. But it's a minor preference of mine to make it more consistent with the rest of the files.